src/Compilers/CSharp/Test/Syntax/Parsing/RefReadonlyTests.cs

```diff
@@ -510,5 +497,859 @@ public void TestNewRefArray()
             EOF();
         }
 
+        [Fact]
+        public void RefReadonly_Declaration()
+        {
+            var source = "void M(ref readonly int p);";
+            UsingDeclaration(source, TestOptions.Regular11);
+            verifyNodes();
+
+            UsingDeclaration(source, TestOptions.RegularNext);
+            verifyNodes();
+
+            UsingDeclaration(source);
+            verifyNodes();
+
+            void verifyNodes()
+            {
+                N(SyntaxKind.MethodDeclaration);
+                {
+                    N(SyntaxKind.PredefinedType);
+                    {
+                        N(SyntaxKind.VoidKeyword);
+                    }
+                    N(SyntaxKind.IdentifierToken, "M");
+                    N(SyntaxKind.ParameterList);
+                    {
+                        N(SyntaxKind.OpenParenToken);
+                        N(SyntaxKind.Parameter);
+                        {
+                            N(SyntaxKind.RefKeyword);
+                            N(SyntaxKind.ReadOnlyKeyword);
+                            N(SyntaxKind.PredefinedType);
+                            {
+                                N(SyntaxKind.IntKeyword);
+                            }
+                            N(SyntaxKind.IdentifierToken, "p");
+                        }
+                        N(SyntaxKind.CloseParenToken);
+                    }
+                    N(SyntaxKind.SemicolonToken);
+                }
+                EOF();
+            }
+        }
+
+        [Fact]
+        public void RefReadonly_Compilation()
```

Comment:
>RefReadonly_Compilation [](http://example.com/codeflow?start=20&length=23)

Semantic diagnostics tests do not belong in parsing tests. Please move them to the file that will be responsible for testing semantics for `ref readonly` parameters. Some legacy tests do that because they used to test for parsing diagnostic, but then the phase reporting the diagnostic got changed and the tests were adjusted in place.

Reply:
I recommend creating a new file in Emit2 project. We are pretty close to reaching the string heap limit in older test projects.